server/filestore_test.go

```diff
@@ -6512,6 +6512,40 @@ func TestFileStoreLargeFullStateMetaCleanup(t *testing.T) {
 
 }
 
+func TestFileStoreIndexDBExistsAfterShutdown(t *testing.T) {
+	sd := t.TempDir()
+	fs, err := newFileStore(
+		FileStoreConfig{StoreDir: sd},
+		StreamConfig{Name: "zzz", Subjects: []string{">"}, Storage: FileStorage})
+	require_NoError(t, err)
+
+	subj, _ := "foo.bar.baz", bytes.Repeat([]byte("ABC"), 33) // ~100bytes
+	for i := 0; i < 1000; i++ {
+		fs.StoreMsg(subj, nil, nil)
+	}
+
+	idxFile := filepath.Join(sd, msgDir, streamStreamStateFile)
+
+	fs.mu.Lock()
+	fs.dirty = 1
+	if err := os.Remove(idxFile); err != nil && !errors.Is(err, os.ErrNotExist) {
+		t.Fatal(err)
+	}
+	fs.mu.Unlock()
+
+	fs.Stop()
+
+	checkFor(t, time.Second, 50*time.Millisecond, func() error {
+		if !fs.closed {
```

Comment:
Need to lock.

Reply:
Actually I think we probably don't even need to check `fs.closed` at all, as `Stop` blocks until it is set. Will remove it.